Sources/Private/MainThread/LayerContainers/MainThreadAnimationLayer.swift

```diff
@@ -149,8 +149,16 @@ final class MainThreadAnimationLayer: CALayer, RootAnimationLayer {
   /// The animatable Current Frame Property
   @NSManaged var currentFrame: CGFloat
 
-  /// The parent `LottieAnimationView` that manages this layer
-  weak var animationView: LottieAnimationView?
+  /// The parent `LottieAnimationLayer` that manages this layer
+  weak var lottieAnimationLayer: LottieAnimationLayer?
+
+  /// Whether or not to use `forceDisplayUpdate()` when rendering each individual frame.
+  ///  - The main thread rendering engine implements optimizations to decrease the amount
+  ///    of properties that have to be re-rendered on each frame. There are some cases
+  ///    where this can result in bugs / incorrect behavior, so we allow it to be disabled.
+  ///  - Forcing a full render on every frame will decrease performance, and is not recommended
+  ///    except as a workaround to a bug in the main thread rendering engine.
+  var forceDisplayUpdateOnEachFrame = false
```

Comment:
In this context, where does `forceDisplayUpdate()` gets called? From the code it seems we are simply passing this bool to `animationLayer.displayWithFrame(frame:, forceUpdates: )`

Reply:
I described it that way because `forceDisplayUpdate()` is a public API but `forceUpdates: true` is not. 

The implementation of `forceDisplayUpdate()` is:

```swift
func forceDisplayUpdate() {
  for animationLayer in animationLayers {
    animationLayer.displayWithFrame(frame: currentFrame, forceUpdates: true)
  }
}
```

and the relevant code for `forceDisplayUpdateOnEachFrame` is:

```swift